### PR TITLE
Add new event in application callback

### PR DESCRIPTION
**Pull request by pvyawaha**
*Issue #, if available:*

*Description of changes:*
This change adds a new event for the callback from OTA library to application so that application can take further actions for cleanup.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by divekarshubham — DISMISSED** [review]

File: source/include/ota.h [hunk 1]
```diff
@@ -177,8 +178,11 @@ typedef uint32_t OtaErr_t;
  * job has failed in some way and should be rejected.
  *
  * @param[in] eEvent An OTA update event from the OtaJobEvent_t enum.
+ *
+ * @param[in] pData Optional data related to the event.
  */
-typedef void (* OtaAppCallback_t)( OtaJobEvent_t eEvent );
+typedef void (* OtaAppCallback_t)( OtaJobEvent_t eEvent,
+                                   void * pData );
```
Comment (dan4thewin) [inline source/include/ota.h:185]: Why are you using a void * here?

Comment (pvyawaha) [reply source/include/ota.h:185]: We have it void ptr so that we application can apply it to any data as the data will be based on the event type.

**Review by dan4thewin — APPROVED** [review]

File: source/ota.c [hunk 2]
```diff
@@ -910,7 +913,7 @@ static void dataHandlerCleanup( IngestResult_t result )
     }
 
     /* Let main application know of our result. */
-    otaAgent.OtaAppCallback( ( result == IngestResultFileComplete ) ? OtaJobEventActivate : OtaJobEventFail );
+    otaAgent.OtaAppCallback( ( result == IngestResultFileComplete ) ? OtaJobEventActivate : OtaJobEventFail, NULL );
```
Comment (yanjos-dev) [inline source/ota.c:916]: I'd prefer if this was with an if else statement.

But if we want to keep the turnery operator, then can you add another set of parenthesis around the turnery operator?

**Review by dan4thewin — APPROVED** [review]

File: source/ota.c (hunk 2 above)
Comment (pvyawaha) [reply source/ota.c:916]: I will help in readability and will be added in next PR.

**Review by divekarshubham — APPROVED** [review]

**Pull request merged**